### PR TITLE
Issue #308: List checks that will miss some violations when only use supportContextStrategyChecks property

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #308: List checks that will miss some violations when only use supportContextStrategyChecks property

please review and merge https://github.com/checkstyle/patch-filters/pull/303 first.

### CODE REVIEW

**Comment by romani** [comment]
@HuGanghui , blocker PR is merged, please rebase this PR and prepare for review.

**Comment by HuGanghui** [comment]
this PR is ok to be reviewed.

File: README.md [hunk 1]
```diff
@@ -74,6 +74,62 @@ Attention: `supportContextStrategyChecks` and `checkNameForContextStrategyByToke
  | neverSuppressedChecks                            | String has user defined TreeWalker Checks to never suppress if files are touched, split by comma. This property is useful for Checks that place violation on whole file or on not all (first/last) occurrence of cause/violated code. | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
  | checkNamesForContextStrategyByTokenOrParentSet | String has user defined TreeWalker Checks that need modify violation nodes to their parent node to expand the context scope, split by comma | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
 
+#### Notes
+
+Currently, the following checks will suppress some violations that should not be suppressed when using `supportContextStrategyChecks`,
```
Comment (rdiachenko) [inline README.md:79]: Put period `.` instead of `,` at the sentence end.

File: README.md [hunk 2]
```diff
@@ -74,6 +74,62 @@ Attention: `supportContextStrategyChecks` and `checkNameForContextStrategyByToke
...
+* ParameterNumberCheck
+* SuperClone
+* SuperFinalize
+* BooleanExpressionComplexity
+* OverloadMethodsDeclarationOrder
+* UnnecessarySemicolonInEnumeration
+
+Also, the following checks will suppress some violations that should not be suppressed when using `supportContextStrategyChecks` and 
+`checkNamesForContextStrategyByTokenOrParentSet`. So you need use `checkNamesForContextStrategyByTokenOrAncestorSet` 
+or `neverSuppressedChecks` to solve this problem:
```
Comment (rdiachenko) [inline README.md:92]: 1. Should be on the same line
2. Did you mean `supportContextStrategyChecks or checkNamesForContextStrategyByTokenOrParentSet` but not `supportContextStrategyChecks and checkNamesForContextStrategyByTokenOrParentSet`?

File: README.md [hunk 3]
```diff
@@ -74,6 +74,62 @@ Attention: `supportContextStrategyChecks` and `checkNameForContextStrategyByToke
...
+* FallThrough
+* RightCurly
+
+Then the following checks will suppress some violations that should not be suppressed unless you use 
+`neverSuppressedChecks`:
+
+
+* VariableDeclarationUsageDistance
+
+* violation node is CLASS_DEF, PACKAGE_DEF or IMPORT
```
Comment (rdiachenko) [inline README.md:110]: How this information is related to `neverSuppressedChecks`?

File: README.md [hunk 4]
```diff
@@ -74,6 +74,62 @@ Attention: `supportContextStrategyChecks` and `checkNameForContextStrategyByToke
...
+    * ImportOrder
+    * RedundantImport
+    * UnusedImports
+    * PackageDeclaration
+    * MissingJavadocPackage
+    * MissingJavadocType
+    * OuterTypeNumberCheck
+    * OuterTypeFilename
+
+* violation information not ast node, just a line information
```
Comment (rdiachenko) [inline README.md:123]: How this information is related to `neverSuppressedChecks`?  Why should we care about ast?

File: README.md [hunk 5]
```diff
@@ -74,6 +74,62 @@ Attention: `supportContextStrategyChecks` and `checkNameForContextStrategyByToke
...
+    * NoCodeInFileCheck
+    * Regexp
+
+* checks belong to Javadoc module do not support ast tree
```
Comment (rdiachenko) [inline README.md:127]: How this information is related to `neverSuppressedChecks`? Why should we care about ast?

**Review by rdiachenko — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (HuGanghui) [reply README.md:92]: 1. done
2. yes, changed.

File: README.md (hunk 3 above)
Comment (HuGanghui) [reply README.md:110]: this information is the reason why these checks need use `neverSuppressedChecks`, but maybe it's not necessary to let users to know it. I have removed it.

File: README.md (hunk 4 above)
Comment (HuGanghui) [reply README.md:123]: replied in https://github.com/checkstyle/patch-filters/pull/309#discussion_r482090170

File: README.md (hunk 5 above)
Comment (HuGanghui) [reply README.md:127]: replied in https://github.com/checkstyle/patch-filters/pull/309#discussion_r482090170

File: README.md (hunk 1 above)
Comment (HuGanghui) [reply README.md]: done

**Review by rdiachenko — APPROVED** [review]
LGTM

**Pull request merged**